### PR TITLE
Fixed various threadpool errors

### PR DESCRIPTION
**Pull request by sergio-nsk**
- double decremented `threadpool->num_threads`
- data race while writing `threadpool->stop` and reading in threads
- attempt to unlock of unlocked mutex

### CODE REVIEW

**Comment by nmoinvaz** [comment]
I think for the double-decrement of `num_threads`, that the decrement should be removed from `threadpool_delete_threads` instead of inside the thread.

**Comment by sergio-nsk** [comment]
I don't think the decrement in a thread function is a good idea by several reasons:

- `threadpool_do_work` can early exit if `thread` is `NULL` and `num_threads` will not decremented.
- `pthread_create` can fail and `num_threads` will be kept incremented.
- decrementing `num_threads` in the threas job function will require using the mutex in `threadpool_delete_threads`

File: threadpool_winxp.c [hunk 1]
```diff
@@ -146,26 +141,25 @@ static uint32_t __stdcall threadpool_do_work(void *arg) {
...
     return 0;
 }
 
 static void threadpool_create_thread_on_demand(threadpool_s *threadpool) {
     // Create new thread and add it to the list of threads
-    threadpool_thread_s *thread = (threadpool_thread_s *)calloc(1, sizeof(threadpool_thread_s));
-    if (!thread)
+    uint32_t id = 0;
+    HANDLE handle = (HANDLE)_beginthreadex(NULL, 0, threadpool_do_work, threadpool, 0, &id);
+    if (handle = -1)
```
Comment (nmoinvaz) [inline threadpool_winxp.c:153]: Should be `== -1`

Comment (sergio-nsk) [reply threadpool_winxp.c:153]: It's impossible, I remember I pressed `=` twice.

**Comment by nmoinvaz** [comment]
Looks like does not pass build.

File: threadpool_pthread.c [hunk 2]
```diff
@@ -140,26 +135,23 @@ static void *threadpool_do_work(void *arg) {
 
     LOG_DEBUG("threadpool - worker 0x%" PRIx64 " - stopped\n", (uint64_t)pthread_self());
 
-    // Reduce thread count
```
Comment (nmoinvaz) [inline threadpool_pthread.c:143]: This did not mean to be removed correct?

File: threadpool_winxp.c [hunk 3]
```diff
@@ -146,26 +141,25 @@ static uint32_t __stdcall threadpool_do_work(void *arg) {
 
     LOG_DEBUG("threadpool - worker 0x%" PRIx32 " - stopped\n", thread->id);
 
-    // Reduce thread count
```
Comment (nmoinvaz) [inline threadpool_winxp.c:149]: This did not mean to be removed correct?

**Review by nmoinvaz — APPROVED** [review]

**Pull request merged**